blocks/library/code-mirror/index.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * External dependencies
+ */
+import CodeMirror from 'react-codemirror';
+import he from 'he';
+require( 'codemirror/mode/css/css' );
+require( 'codemirror/mode/diff/diff' );
+require( 'codemirror/mode/javascript/javascript' );
+require( 'codemirror/mode/markdown/markdown' );
+require( 'codemirror/mode/pegjs/pegjs' );
+require( 'codemirror/mode/php/php' );
+require( 'codemirror/lib/codemirror.css' );
+
+/**
+ * Internal dependencies
+ */
+import { registerBlockType, query } from '../../api';
+const { html } = query;
+
+const toEditor = s => he.decode( s.split( '<br>' ).join( '\n' ) );
+const fromEditor = s => he.encode( s, { useNamedReferences: true } ).split( '\n' ).join( '<br>' );
+
+registerBlockType( 'core/code-mirror', {
+	title: wp.i18n.__( 'Code Editor' ),
+	icon: 'text',
+	category: 'common',
+	content: 'just a test',
+
+	attributes: {
+		html: html(),
+	},
+
+	edit( { attributes, setAttributes } ) {
+		return (
+			<div>
+				<select onBlur={ ( { target: { value } } ) => setAttributes( { language: value } ) }>
+					{ [ 'css', 'diff', 'javascript', 'markdown', 'pegjs', 'php' ].map( language => (
+						<option
+							key={ language }
+							selected={ language === attributes.language }
```

Comment:
thanks. I honestly didn't pay any attention to this (obviously). it seems like I need to make it a setting in the page inspector too but at the time I didn't know about that